app/src/lib/git/checkout.ts

```diff
@@ -110,6 +117,72 @@ export async function checkoutBranch(
   return true
 }
 
+/**
+ * Check out the given commit.
+ * Literally invokes `git checkout <commit SHA>`.
+ *
+ * @param repository - The repository in which the branch checkout should
+ *                     take place
+ *
+ * @param commit     - The commit that should be checked out
+ *
+ * @param progressCallback - An optional function which will be invoked
+ *                           with information about the current progress
+ *                           of the checkout operation. When provided this
+ *                           enables the '--progress' command line flag for
+ *                           'git checkout'.
+ */
+export async function checkoutCommit(
+  repository: Repository,
+  account: IGitAccount | null,
+  commit: CommitOneLine,
+  progressCallback?: ProgressCallback
+): Promise<true> {
+  let opts: IGitExecutionOptions = {
+    env: await envForRemoteOperation(
+      account,
+      getFallbackUrlForProxyResolve(account, repository)
+    ),
+    expectedErrors: AuthenticationErrors,
+  }
+
+  if (progressCallback) {
+    const title = `Checking out commit ${commit.sha}`
+    const kind = 'checkout'
+    const targetCommit = commit.sha
```

Comment:
```suggestion
    const target = commit.sha
```

Similar minor commit to above, if this is target, than below becomes `value, target})`